oceans/sw_extras/sw_extras.py

```diff
@@ -615,13 +615,13 @@ def spice(s, t, p):
 
     sp = np.zeros_like(pt)
     T = np.ones_like(pt)
-    s -= 35
+    s_ = s - 35
```

Comment:
You are correct. That happens if the input is a `numpy` array b/c `asanyarray` does not perform a copy.

I believe you don't need to different var name here. Just change the in-place operation to `s = s - 35`. And then everything should happen in the scope of the function.